components/director/internal/domain/application/resolver.go

```diff
@@ -238,6 +254,62 @@ func (r *Resolver) Application(ctx context.Context, id string) (*graphql.Applica
 	return r.appConverter.ToGraphQL(app), nil
 }
 
+// ApplicationForTenant could be executed only in a certificate flow. It fetches model.ApplicationTemplate by label scenarioassignment.SubaccountIDKey
+// with value equal to the consumer ID. Lists all tenant header scoped []model.Application and finds the one created by the previously fetched App Template
+func (r *Resolver) ApplicationForTenant(ctx context.Context) (*graphql.Application, error) {
```

Comment:
As discussed w/t the team - we want the logic of this function to be actually embedded in the existing `applications` query so that we don't introduce new standalone queries unnecessarily.